src/storage/storage.ts

```diff
@@ -354,6 +354,7 @@ export class Storage implements IStorage {
    * @returns {Promise<void>}
    */
   async processHistory(): Promise<void> {
+    await this.store.preProcess();
     await processHistory(this, { rewardLock: this.version?.reward_spend_min_blocks });
```

Comment:
Is this a recursion? I got confused by the name of the method.

Reply:
The `processHistory` is the name of the method and an util from `src/utils/storage.ts`.
Since the implementation requires other imports having it here creates some circular dependencies.

This also makes it easier to test the history processing.

Reply:
Got it. I guess this is very error prone. Should we rename one of these? Maybe just rename during the import phase and call it `storageProcessHistory`?

Reply:
Renamed the util to `processHistoryUtil`

Reply:
Thanks!